python/src/triton.cc

```diff
@@ -1106,7 +1106,20 @@ void init_triton_ir(py::module &&m) {
               mlir::Value &ptr, mlir::Value &val,
               mlir::Value &mask) -> mlir::Value {
              auto loc = self.getUnknownLoc();
-             mlir::Type dstType = val.getType();
+             auto srcTensorType =
```

Comment:
You can try `if (auto srcTensorType = ptr.getType().dyn_cast<mlir::RankedTensorType>()) { ... }`.
This will help hide srcTensorType as a local variable.
So is others.

Reply:
Done